lib/Runtime/Library/TypedArray.cpp

```diff
@@ -606,15 +606,39 @@ namespace Js
             object;
     };
 
-    BOOL TypedArrayBase::HasProperty(PropertyId propertyId)
+    BOOL TypedArrayBase::HasOwnProperty(PropertyId propertyId)
+    {
+        return JavascriptConversion::PropertyQueryFlagsToBoolean(HasPropertyQuery(propertyId));
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(PropertyId propertyId, ScriptContext *scriptContext)
+    {
+        PropertyString *propertyString = scriptContext->GetPropertyString(propertyId);
+        return CanonicalNumericIndexString(propertyString, scriptContext);
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(JavascriptString *propertyString, ScriptContext *scriptContext)
+    {
+        double result;
+        return JavascriptConversion::CanonicalNumericIndexString(propertyString, &result, scriptContext);
+    }
+
+    PropertyQueryFlags TypedArrayBase::HasPropertyQuery(PropertyId propertyId)
     {
         uint32 index = 0;
-        if (GetScriptContext()->IsNumericPropertyId(propertyId, &index) && (index < this->GetLength()))
+        ScriptContext *scriptContext = GetScriptContext();
+        if (scriptContext->IsNumericPropertyId(propertyId, &index))
         {
             // All the slots within the length of the array are valid.
-            return true;
+            return index < this->GetLength() ? Property_Found : Property_NotFound_NoProto;
```

Comment:
Good idea, although I found 131 occurrences in 24 files. Since this is not a new method name introduced in this changeset, I would leave it for future cosmetic PRs.

---
In reply to: [106311219](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106311219) [](ancestors = 106311219)